converter/src/main/java/gov/cms/qpp/conversion/validate/CpcQualityMeasureSectionValidator.java

```diff
@@ -0,0 +1,101 @@
+package gov.cms.qpp.conversion.validate;
+
+import gov.cms.qpp.conversion.model.Node;
+import gov.cms.qpp.conversion.model.Program;
+import gov.cms.qpp.conversion.model.TemplateId;
+import gov.cms.qpp.conversion.model.Validator;
+import gov.cms.qpp.conversion.model.error.ErrorCode;
+import gov.cms.qpp.conversion.model.error.LocalizedError;
+import gov.cms.qpp.conversion.model.validation.MeasureConfig;
+import gov.cms.qpp.conversion.model.validation.MeasureConfigs;
+
+import java.util.Arrays;
+import java.util.List;
+import java.util.Map;
+
+/**
+ * Validates a measure groupings for a CPC+ Quality Measure Section node.
+ */
+@Validator(value = TemplateId.MEASURE_SECTION_V2, program = Program.CPC)
+public class CpcQualityMeasureSectionValidator extends NodeValidator {
+
+	/**
+	 * Validate that the Quality Measure Section contains an acceptable combination of measures...
+	 *
+	 * @param node The node to validate.
+	 */
+	@Override
+	protected void internalValidateSingleNode(Node node) {
+		Checker checker = check(node);
+
+		Arrays.stream(CpcGroupMinimum.values())
+				.forEach(group -> checkGroupMinimum(checker, group));
+		verifyOverallCount(checker);
+	}
+
+	/**
+	 * Verify that CPC+ measurement group minimums are met.
+	 * @param checker node validator helper
+	 * @param groupMinimum group minimum config
+	 */
+	private void checkGroupMinimum(Checker checker, CpcGroupMinimum groupMinimum) {
+		String[] measureIds = grabGroupMeasures(groupMinimum);
+		checker.hasMeasures(
+				groupMinimum.makeError(measureIds), groupMinimum.minimum, measureIds);
+	}
+
+	/**
+	 * Retrieve measure ids for group specific measures.
+	 * @param groupMinimum group config
+	 * @return measure id array
+	 */
+	String[] grabGroupMeasures(CpcGroupMinimum groupMinimum) {
+		Map<String, List<MeasureConfig>> cpcPlusGroups = MeasureConfigs.getCpcPlusGroups();
+
+		return cpcPlusGroups.get(groupMinimum.name()).stream()
+				.map(MeasureConfig::getElectronicMeasureVerUuid)
+				.toArray(String[]::new);
+	}
+
+	/**
+	 * Verify minimum across all groups.
+	 * @param checker node validator helper
+	 */
+	private void verifyOverallCount(Checker checker) {
+		String[] measureIds = MeasureConfigs.getCpcPlusGroups()
+				.values().stream()
+				.flatMap(List::stream)
+				.map(MeasureConfig::getElectronicMeasureVerUuid)
+				.toArray(String[]::new);
+
+		checker.hasMeasures(
+				CpcGroupMinimum.makeOverallError(measureIds), CpcGroupMinimum.NUMBER_OF_MEASURES_REQUIRED, measureIds);
+	}
+
+	/**
+	 * A holder of CPC+ group specific configuration information.
+	 */
+	enum CpcGroupMinimum {
+		A("outcome", 2),
+		B("complex process", 2);
+
+		private static int NUMBER_OF_MEASURES_REQUIRED = 9;
```

Comment:
![MAJOR](https://sonarsource.github.io/sonar-github/severity-major.png 'Severity: MAJOR') Assign this magic number 9 to a well-named constant, and use the constant instead. [![rule](https://sonarsource.github.io/sonar-github/rule.png)](https://sonarcloud.io/coding_rules#rule_key=squid%3AS109)
![MINOR](https://sonarsource.github.io/sonar-github/severity-minor.png 'Severity: MINOR') Rename this field "NUMBER_OF_MEASURES_REQUIRED" to match the regular expression '^[a-z][a-zA-Z0-9]*$'. [![rule](https://sonarsource.github.io/sonar-github/rule.png)](https://sonarcloud.io/coding_rules#rule_key=squid%3AS3008)